### PR TITLE
Merge pull request #49 from towerofnix/sound-objects

### PR DESCRIPTION
**Pull request by towerofnix**
...and match Scratch behavior with regards to restarting sounds, dealing with yields, etc.

Alongside sb-edit#21, this implements playing sounds from exported Scratch 3.0 projects. (No WAV support yet though - that should be implemented separately.)

Test it out with this project: https://scratch.mit.edu/projects/360677530/

### CODE REVIEW

**Comment by PullJosh** [comment]
> Test it out with this project

Loving the trend of providing a test project. Thanks!

File: scratch-js/Project.mjs [hunk 1]
```diff
@@ -156,28 +156,6 @@ export default class Project {
...
-
-      this.playingSounds.push(sound);
-
-      audio.play().then(() => {
-        sound.hasStarted = true;
-      });
-    });
-  }
-
   _stopSound(sound) {
```
Comment (PullJosh) [inline scratch-js/Project.mjs]: `_stopSound` can be safely deleted, right?

Comment (towerofnix) [reply scratch-js/Project.mjs]: Oh yup! Thanks. Pushed!

**Comment by towerofnix** [comment]
>Loving the trend of providing a test project. Thanks!

Happy to make them! Not only are they good for confirming these PRs do what they're supposed to, they're also good ways to make sure I haven't missed any behavior, and they serve as good test projects for any future work with Scratch interpreters/compilers too.

**Comment by PullJosh** [comment]
This PR was an excellent one. Thank you @towerofnix!

**Pull request merged**